File: sangsoo/0x0C Backtracking/15649.py
N, M = map(int, input().split())
arr = [0 for _ in range(M)]
isused = [0 for _ in range(N)]
def func(k:int)->None:
    if (k == M):
        for i in range(M):
            print(arr[i], end = ' ')
        print("")
        return # return 안 넣어주면 에러
    for i in range(N):
        if(not isused[i]):
            isused[i] = 1
            arr[k] = i+1
            func(k+1)
            isused[i] = 0
func(0)

File: sangsoo/0x0C Backtracking/15649.cpp
#include <iostream>
// N과 M (1) Permutation
using namespace std;
int N, M;
int arr[10];
int isused[10];
void func(int k){
    if (k == M){
        for(int i = 0; i < M; i++)
            cout << arr[i] << ' ';
        cout << '\n';
    }
    for (int i = 0; i < N; i++){
        if(!isused[i]){
            isused[i] = 1;
            arr[k] = i+1;
            func(k+1);
            isused[i] = 0;
        }
    }

}

int main(){
    ios::sync_with_stdio(0);
    cin.tie(0);

    cin >> N >> M;
    func(0);

}